src/main/java/de/tum/in/www1/artemis/service/connectors/gitlab/dto/GitLabCommitDTO.java

```diff
@@ -10,111 +10,17 @@
 
 @JsonIgnoreProperties(ignoreUnknown = true)
 @JsonInclude(JsonInclude.Include.NON_EMPTY)
-public class GitLabCommitDTO {
+public record GitLabCommitDTO(@JsonProperty("id") String hash, String message, ZonedDateTime timestamp, @JsonProperty("url") String commitUrl, Author author, List<String> added,
+        List<String> modified, List<String> removed) {
 
-    @JsonProperty("id")
-    private String hash;
-
-    private String message;
-
-    private ZonedDateTime timestamp;
-
-    @JsonProperty("url")
-    private String commitUrl;
-
-    private Author author;
-
-    private List<String> added = new ArrayList<>();
-
-    private List<String> modified = new ArrayList<>();
-
-    private List<String> removed = new ArrayList<>();
-
-    public String getHash() {
-        return hash;
-    }
-
-    public void setHash(String hash) {
-        this.hash = hash;
-    }
-
-    public String getMessage() {
-        return message;
-    }
-
-    public void setMessage(String message) {
-        this.message = message;
-    }
-
-    public ZonedDateTime getTimestamp() {
-        return timestamp;
-    }
-
-    public void setTimestamp(ZonedDateTime timestamp) {
-        this.timestamp = timestamp;
-    }
-
-    public String getCommitUrl() {
-        return commitUrl;
-    }
-
-    public void setCommitUrl(String commitUrl) {
-        this.commitUrl = commitUrl;
-    }
-
-    public Author getAuthor() {
-        return author;
-    }
-
-    public void setAuthor(Author author) {
-        this.author = author;
-    }
-
-    public List<String> getAdded() {
-        return added;
-    }
-
-    public void setAdded(List<String> added) {
-        this.added = added;
-    }
-
-    public List<String> getModified() {
-        return modified;
-    }
-
-    public void setModified(List<String> modified) {
-        this.modified = modified;
-    }
-
-    public List<String> getRemoved() {
-        return removed;
-    }
-
-    public void setRemoved(List<String> removed) {
-        this.removed = removed;
+    // make sure the lists are never null
+    public GitLabCommitDTO {
+        added = (added != null) ? added : new ArrayList<>();
+        modified = (modified != null) ? modified : new ArrayList<>();
+        removed = (removed != null) ? removed : new ArrayList<>();
```

Comment:
We should probably adjust the code utilizing this DTO to also accept empty Arrays (if it makes sense), since all other DTOs also send null values for empty collections.

Reply:
I couldn't find a place where we access these lists explicitly and the tests succeed without setting these default values. However, I am not familiar with the GitLab API and don't know if we can just remove the code. Maybe @krusche can provide more information

Reply:
Tested on ts5. Works with empty commits after the code was removed